File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        
        HashSet<ListNode> set = new HashSet<>();

        ListNode currB = headB;

        while(currB!=null){
            set.add(currB);
            currB=currB.next;
        }

        ListNode currA = headA;

        while(currA!=null){
            if(set.contains(currA)){
                return currA;
            }
            currA = currA.next;
        }
        return null;
        
    }
}

File: 160-intersection-of-two-linked-lists/intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
// create a set
    let set = new Set();

    
// all the headB added into the set
    while(headB){
        set.add(headB);
        headB=headB.next;
    }   
// check headA present in the set return the element 
    while(headA){
        if(set.has(headA)){
            return headA;
        }
        headA=headA.next;
    }
    return null;  
};